databricks/koalas/series.py

```diff
@@ -4772,20 +4772,49 @@ def dot(self, other) -> Union[Scalar, "Series"]:
 
         >>> s @ s
         14
+
+        >>> kdf = ks.DataFrame({'x': [0, 1, 2, 3], 'y': [0, -1, -2, -3]})
+        >>> kdf
+           x  y
+        0  0  0
+        1  1 -1
+        2  2 -2
+        3  3 -3
+
+        >>> with ks.option_context("compute.ops_on_diff_frames", True):
+        ...     s.dot(kdf)
+        ...
+        x    14
+        y   -14
+        dtype: int64
         """
         if isinstance(other, DataFrame):
-            raise ValueError(
-                "Series.dot() is currently not supported with DataFrame since "
-                "it will cause expansive calculation as many as the number "
-                "of columns of DataFrame"
-            )
-        if self._kdf is not other._kdf:
-            if len(self.index) != len(other.index):
+            if not same_anchor(self, other) and not self.index.sort_values().equals(
```

Comment:
Ah .. this will be very expensive .. but I have no better idea. But we'll have to fail fast if `compute.ops_on_diff_frames` is disabled. Currently, it looks it's going to sort each DataFrame first and then throw an exception.

Reply:
Makes sense! Modified to fail fast here.

Reply:
Do we actually need to raise the exception here?
The `sort_values()` is just a planning, so I guess it's not a big deal.
We should rather fix the `Index.equals` which I will work on as a while Index operations.

Reply:
I see. Removed the exception. Thanks!